tap_postgres/tap.py

```diff
@@ -5,31 +5,90 @@
 import io
 import signal
 from functools import cached_property
-from typing import TYPE_CHECKING, Any
+from typing import Any, Mapping, cast
 
 import paramiko
 from singer_sdk import SQLTap, Stream
 from singer_sdk import typing as th  # JSON schema typing helpers
+from sqlalchemy.engine import URL
 from sqlalchemy.engine.url import make_url
 from sshtunnel import SSHTunnelForwarder
 
 from tap_postgres.client import PostgresConnector, PostgresStream
 
-if TYPE_CHECKING:
-    from sqlalchemy.engine.url import URL
-
 
 class TapPostgres(SQLTap):
     """Singer tap for Postgres."""
 
     name = "tap-postgres"
     default_stream_class = PostgresStream
 
+    def __init__(
+        self,
+        *args,
+        **kwargs,
+    ) -> None:
+        """Constructor.
+
+        Should use JSON Schema instead
+        See https://github.com/MeltanoLabs/tap-postgres/issues/141
+        """
+        super().__init__(*args, **kwargs)
+        assert (self.config.get("sqlalchemy_url") is not None) or (
+            self.config.get("host") is not None
+            and self.config.get("port") is not None
+            and self.config.get("user") is not None
+            and self.config.get("password") is not None
+        ), (
+            "Need either the sqlalchemy_url to be set or host, port, user,"
+            + " and password to be set"
+        )
+
     config_jsonschema = th.PropertiesList(
+        th.Property(
+            "host",
+            th.StringType,
+            description=(
+                "Hostname for postgres instance. "
+                + "Note if sqlalchemy_url is set this will be ignored."
+            ),
+        ),
+        th.Property(
+            "port",
+            th.IntegerType,
+            default=5432,
+            description=(
+                "The port on which postgres is awaiting connection. "
+                + "Note if sqlalchemy_url is set this will be ignored."
+            ),
+        ),
+        th.Property(
+            "user",
+            th.StringType,
+            description=(
+                "User name used to authenticate. "
+                + "Note if sqlalchemy_url is set this will be ignored."
+            ),
+        ),
+        th.Property(
+            "password",
+            th.StringType,
```

Comment:
```suggestion
            th.StringType,
            secret=True,
```

@visch I think I added a similar comment to another PR. I dont think it really makes any functional difference but it helps with how it advertises its metadata to things like meltano and the hub 👍 .